File: Test/CPP/CompTests/UnitConversionConstantTest.cpp
/////////////////////////////////////////////////////////////////////////////////
//2345678901234567890123456789012345678901234567890123456789012345678901234567890
/////////////////////////////////////////////////////////////////////////////////
// PROJECT: Unit Conversion
//
// Copyright Copyright 2024 MAP
//
// Unpublished - Rights reserved under the Copyright Laws of the United States
//
/////////////////////////////////////////////////////////////////////////////////
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
// 
// This program is distributed in the hope that it will be useful, 
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program; if not, write to the Free Software
// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
//
/////////////////////////////////////////////////////////////////////////////////
//
// File UnitConversionConstantTest.cpp
//
// Program to test constant methods. 
// 
// Version 1.0
//
/////////////////////////////////////////////////////////////////////////////////
#include "UnitConversionConstantTest.hpp"
#include "Constants.hpp"
#include "Version.hpp"
#include <iostream>

/** 
 * @brief Constructor
 *
 * @param[in] brief - brief output of test results.
 * @param[in] path  - the path to the output folder where the
 *                    test results are written.
 */
UnitConversionConstantTest::UnitConversionConstantTest(bool brief, 
                                                       const std::string& path) 
                                                       : UnitConversionBaseTest(brief, path)
{
    openFile("ConstantTest.txt");
}

/**
 * @brief Write constant to output stream.
 *
 * @param[in] constant - constant to output.
 * @param[in] name     - constant name.
 * @param[in] out      - stream to write output to.
 *
 */
void UnitConversionConstantTest::printConstant(const ConstantGroup& constant, 
                                               const std::string& name, 
                                               std::ostream& out)
{
    std::list<std::string> unitNames = constant.constantNames();
    printList(unitNames, "Constants", name,out);
    for (std::list<std::string>::iterator it = unitNames.begin(); 
                                          it != unitNames.end(); ++it)
    {
        UBASE u    = constant.constant(*it);
        double val = constant.value(*it);

        printResult(u.valid(), name + " " + *it, "valid",
                    bool_to_str(u.valid()) +", "+ u.toString(), 
                    "true",out);
    }
    out << std::endl;
}

/*
 * @brief Run the tests; output results to screen and file.
 * 
 * @param[in] out - stream to write to.
 */
void UnitConversionConstantTest::runOut(std::ostream& out)
{
    clock_t start = clock();
    printHeader("UnitConversionConstantTest", out);

    Constants& constants = Constants::Instance();

    std::list<std::string> constantNames = constants.names();
    printList(constantNames, "Constants", "CONSTANTS",out);

    for (std::list<std::string>::iterator it = constantNames.begin(); 
                                          it != constantNames.end(); ++it)
    {
        if (*it == "Invalid")
        {
            continue;
        }
        else
        {
            printConstant(constants.constant(*it), *it,out);
        }
    }

    clock_t end = clock();
    double diff = double(end - start)/CLOCKS_PER_SEC;
    printFooter("UnitConversionConstantTest", diff, out);
}
// EOF

File: Test/CS/UnitConversionTest/UnitConversionTest/UnitConversionConstantTest.cs
/////////////////////////////////////////////////////////////////////////////////
//2345678901234567890123456789012345678901234567890123456789012345678901234567890
/////////////////////////////////////////////////////////////////////////////////
// PROJECT: Unit Conversion
//
// Copyright Copyright 2024 MAP
//
// Unpublished - Rights reserved under the Copyright Laws of the United States
//
/////////////////////////////////////////////////////////////////////////////////
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
// 
// This program is distributed in the hope that it will be useful, 
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program; if not, write to the Free Software
// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
//
/////////////////////////////////////////////////////////////////////////////////
//
// File UnitConversionConstantTest.cs
//
// Program to test constant methods. 
// 
// Version 1.0
//
/////////////////////////////////////////////////////////////////////////////////
namespace UnitConversionTestCS
{
    using System.Collections.Generic;
    using System;
    using UnitConversion;

    /// <summary>
    /// Program to test constant methods.
    /// </summary>
    public class UnitConversionConstantTest : UnitConversionBaseTest
    {
        /// <summary>
        /// Constructor. 
        /// </summary>
        /// <param><c>brief</c> (input)  brief output for test results.</param>
        /// <param><c>path</c>  (input) path to directory for output 
        ///                             files.</param>
        public UnitConversionConstantTest(bool brief, string path) : base(brief, path)
        {
             openFile("ConstantTest.txt");
        }

        /// <summary>
        /// Write constant to output stream. 
        /// </summary>
        /// <param><c>constant</c> (input) constant to output.</param>
        /// <param><c>name</c>     (input) constant name.</param>
        private void printConstant(ConstantGroup constant, string name)
        {
            List<string> unitNames = constant.constantNames();
            printList(unitNames,   "Constants", name);
            foreach(string it in unitNames)
            {
               UBASE u = constant.constant(it);
               double val = constant.value(it);
               printResult(u.valid(), name + " " + it, "valid",
                           bool_to_str(u.valid()) + ", " + u.toString(),
                           "true");
            }
            Console.WriteLine(" ");
        }

        /// <summary>
        /// Run tests; output results to screen or file.
        /// </summary>
        protected override void runOut()
        {
            DateTime start = DateTime.Now;
            printHeader("UnitConversionConstantTest");

            Constants constants =  Constants.Instance();

           List<string> constantNames = constants.names();
           printList(constantNames,   "Constants", "CONSTANTS");
           
           foreach (string it in constantNames)
           {
               if( it == "Invalid")
               {
                   continue;
               }
               else
               {
                   printConstant(constants.constant(it), it);
               }
           }
            DateTime end = DateTime.Now;
            TimeSpan ts = end - start;
            printFooter("UnitConversionConstantTest", ts);
            Console.WriteLine(" ");
        }
    }
}
// EOF
